packages/back-end/src/controllers/organizations.ts

```diff
@@ -865,3 +865,36 @@ export async function putUpload(req: Request, res: Response) {
     status: 200,
   });
 }
+
+export async function putAdminResetUserPassword(
+  req: AuthRequest<{
+    userToUpdateId: string;
+    updatedPassword: string;
+  }>,
+  res: Response
+) {
+  const { updatedPassword } = req.body;
+  const userToUpdateId = req.params.id;
+
+  if (IS_CLOUD) {
+    throw new Error(
+      "This functionality is not available with GrowthBook Cloud"
+    );
+  }
+
+  const isAdmin = req.permissions.organizationSettings;
+  const { org } = getOrgFromReq(req);
+  const isUserToUpdateInSameOrg = org.members.find(
+    (member) => member.id === userToUpdateId
+  );
+
+  // Only update the password if the request came from an admin, and the member
+  // we're updating is in the same org as the requester
+  if (isUserToUpdateInSameOrg && isAdmin) {
```

Comment:
This check might not be necessary, but I kept it in for now as it's a nice safe-guard. But if we deem it unnecessary I can easily rip it out.

Reply:
Updated!